### PR TITLE
Always show the "Select" button in "sonata_type_model_list"

### PR DESCRIPTION
**Pull request by core23**
Rebase of https://github.com/sonata-project/SonataAdminBundle/pull/3252

### Changelog

```markdown
### Changed
- The `Select` button is always visible and has a primary check style in `sonata_type_model_list` popups
```

### Subject
The `Select` button is always visible and never hidden. This is a improvement for mobile devices.

Before
bildschirmfoto 2016-05-29 um 08 59 52

After
bildschirmfoto 2016-05-30 um 16 55 03

### CODE REVIEW

**Comment by ju1ius** [comment]
> This is a improvement for mobile devices.

Not only for mobile! This piece of UI is a bit confusing, because it looks almost the same as the normal list view. In my experience, on their first try, users almost always hesitate about what to do.

Would you mind also changing [the button's class](https://github.com/sonata-project/SonataAdminBundle/blob/master/Resources/views/CRUD/list__select.html.twig#L15) to `btn-primary` ? Since  this is the primary action, it makes sense to make it clearly stand out.

**Comment by core23** [comment]
Good point, changed the button class too @ju1ius

**Comment by Soullivaneuh** [comment]
@core23 Can we have a before/after screenshot?

> This is a improvement for mobile devices.

And a mobile view screenshot? :wink:

File: Resources/views/CRUD/list__select.html.twig [hunk 1]
```diff
@@ -12,7 +12,7 @@ file that was distributed with this source code.
 {% extends admin.getTemplate('base_list_field') %}
 
 {% block field %}
-    <a class="btn btn-default" href="{{ admin.generateObjectUrl('edit', object) }}">
+    <a class="btn btn-primary" href="{{ admin.generateObjectUrl('edit', object) }}">
```
Comment (Soullivaneuh) [inline Resources/views/CRUD/list__select.html.twig]: This is not really part of this PR.

**Comment by core23** [comment]
> And a mobile view screenshot? 😉 

The mobile screenshot is quite the same. The usability is just improved on mobile devices. You can select an entity with only one click instead of two.

File: Resources/views/CRUD/list__select.html.twig (hunk 1 above)
Comment (core23) [inline Resources/views/CRUD/list__select.html.twig]: Updated the changelog.

**Comment by OskarStark** [comment]
LGTM 👍

**Comment by OskarStark** [comment]
Can you please check StyleCi @core23 ?

**Comment by core23** [comment]
Don't know why StyleCI got problems with the existing code. Rebased to the latest HEAD.

@Soullivaneuh Can you say something? This applies to the other bundles too

**Comment by ju1ius** [comment]
A little detail: the icon class for the select button should be [`fa fa-check`](http://fontawesome.io/icon/check/), since it conveys the "select me" intent better than an arrow (which implies something link "go to").

**Comment by ju1ius** [comment]
Another improvement: the select button should replace the batch checkbox (in table mode).
Because:
* the batch checkbox is not used in select mode, so it just confuses users.
* on small screens, you currently have to scroll horizontally to be able to click the select button.

Unfortunately I don't think it is feasible in a BC way. But nice to keep in mind for later !

**Comment by greg0ire** [comment]
Rebasing should fix the CI problem.

**Comment by OskarStark** [comment]
> A little detail: the icon class for the select button should be fa fa-check, since it conveys the "select me" intent better than an arrow (which implies something link "go to").

👍

**Comment by OskarStark** [comment]
Thank you @core23 👍

**Comment by machei** [comment]
`An exception has been thrown during the rendering of a template ("Warning: Missing argument 2 for Sonata\AdminBundle\Admin\AbstractAdmin::generateObjectUrl()") in SonataAdminBundle:CRUD:list__select.html.twig at line 15.` 

Replace into this code:
```
 <a class="btn btn-default" href="{{ admin.generateObjectUrl('list', object) }}">
    <i class="fa fa-arrow-right"></i>
    {{ 'list_select'|trans({}, 'SonataAdminBundle') }}
  </a>
```

**Comment by OskarStark** [comment]
From where is this code?

in this PR we have this:
```diff
-    <a class="btn btn-default" href="{{ admin.generateUrl('list') }}">
 -        <i class="fa fa-arrow-right"></i>
 +    <a class="btn btn-primary" href="{{ admin.generateUrl('list') }}">
 +        <i class="fa fa-check"></i>
          {{ 'list_select'|trans({}, 'SonataAdminBundle') }}
      </a>
```

**Comment by greg0ire** [comment]
@machei : what does this have to do with anything ?

**Comment by machei** [comment]
@greg0ire 

When i am trying to use `sonata_type_model_list` in field, that error still keeps with above message. 

Specific file: https://github.com/core23/SonataAdminBundle/blob/a9ae182b01133a278dd3151d5efcbe77ef926416/Resources/views/CRUD/list__select.html.twig

that code should works fine:
`
-    <a class="btn btn-primary" href="{{ admin.generateUrl('list') }}">
+    <a class="btn btn-primary" href="{{ admin.generateUrl('list', object) }}">
         <i class="fa fa-check"></i>
         {{ 'list_select'|trans({}, 'SonataAdminBundle') }}
     </a>
`

**Comment by OskarStark** [comment]
your solution looks wrong to me, please check: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Admin/AbstractAdmin.php#L1150-L1166

**Comment by greg0ire** [comment]
Did you mean `generateObjectUrl`?

**Comment by machei** [comment]
Very strange...

Why i got error with `generateObjectUrl` instead of generateUrl ?

**Comment by greg0ire** [comment]
You should check your version of the code, and open an issue.

**Comment by OskarStark** [comment]
old cache?
